### PR TITLE
SWC-5733

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/provenance/ProvUtils.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public static ProvGraph buildProvGraph(Map<Reference, String> generatedByActivit
 		// create Expand nodes and add edges to used nodes if requested
 		for (Activity act : processedActivities.values()) {
 			ActivityType type = ActivityTypeUtil.get(act);
-			ActivityGraphNode activityNode = new ActivityGraphNode(createUniqueNodeId(), act.getId(), act.getName(), type, act.getModifiedBy(), act.getModifiedOn(), false);
+			ActivityGraphNode activityNode = new ActivityGraphNode(createUniqueNodeId(), act.getId(), act.getName(), act.getDescription(), type, act.getModifiedBy(), act.getModifiedOn(), false);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/provenance/ProvUtils.java:63]: Pass the activity description to this ActivityGraphNode object.

File: src/main/java/org/sagebionetworks/web/client/widget/provenance/ProvViewUtil.java [hunk 2]
```diff
@@ -78,19 +81,26 @@ public static ProvNodeContainer createActivityContainer(ActivityGraphNode node,
...
+			if (DisplayUtils.isDefined(node.getActivityDescription())) {
+				Paragraph descriptionParagraph = new Paragraph(DisplayUtils.stubStrPartialWord(node.getActivityDescription(), MAX_ACT_CODE_NAME_CHAR));
+				descriptionParagraph.addStyleName(PROV_ACTIVITY_LABEL_STYLE);
+				descriptionParagraph.addStyleName(ACT_MARGIN_TIME);
+				Tooltip description = new Tooltip(descriptionParagraph, node.getActivityDescription());
+				content.add(description);
+			}
 			content.add(time);
 			container.addContent(content);
+			badge.configure(node.getModifiedBy());
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/provenance/ProvViewUtil.java:103]: Do not configure the UserBadge until after it's been attached to the parent (was not rendering).

File: src/main/java/org/sagebionetworks/web/client/widget/provenance/ProvViewUtil.java [hunk 3]
```diff
@@ -78,19 +81,26 @@ public static ProvNodeContainer createActivityContainer(ActivityGraphNode node,
...
-			Widget userBadgeWidget = badge.asWidget();
-			userBadgeWidget.addStyleName(ACT_MARGIN_USER);
-			content.add(userBadgeWidget);
+			Div badgeContainer = new Div();
+			badgeContainer.addStyleName(ACT_MARGIN_USER);
+			badgeContainer.add(badge.asWidget());
+			content.add(badgeContainer);
 			time.addStyleName(ACT_MARGIN_TIME);
+			if (DisplayUtils.isDefined(node.getActivityDescription())) {
+				Paragraph descriptionParagraph = new Paragraph(DisplayUtils.stubStrPartialWord(node.getActivityDescription(), MAX_ACT_CODE_NAME_CHAR));
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/provenance/ProvViewUtil.java:90]: If the activity description is not null and has greater than 0 length, then create a paragraph element with the truncated description, and wrap it with a Tooltip (that has the full description text).

File: src/main/java/org/sagebionetworks/web/shared/provenance/ActivityGraphNode.java [hunk 4]
```diff
@@ -8,17 +8,19 @@
...
 	private String activityName;
+	private String activityDescription;
 	private ActivityType type;
 	private String modifiedBy;
 	private Date modifiedOn;
 
 	public ActivityGraphNode() {}
 
-	public ActivityGraphNode(String id, String activityId, String activityName, ActivityType type, String modifiedBy, Date modifiedOn, Boolean startingNode) {
+	public ActivityGraphNode(String id, String activityId, String activityName, String activityDescription, ActivityType type, String modifiedBy, Date modifiedOn, Boolean startingNode) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/shared/provenance/ActivityGraphNode.java:18]: Not entirely sure why all of these serializable graph node objects exist (why we do not just reference the EntityHeader or Activity), but they're still in the codebase.

**Review by xschildw — APPROVED** [review]

**Pull request merged**